naga/src/back/msl/mod.rs

```diff
@@ -220,6 +220,40 @@ impl Default for Options {
     }
 }
 
+/// A mapping of vertex buffers and their attributes to shader
+/// locations.
+#[derive(Debug, Default, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize))]
+#[cfg_attr(feature = "deserialize", derive(serde::Deserialize))]
+pub struct AttributeMapping {
+    /// Shader location associated with this attribute
+    pub id: u32,
+    /// Offset in bytes from start of vertex buffer structure
+    pub offset: u32,
+    /// Format code to help us unpack the attribute into the type
+    /// used by the shader. Codes correspond to a 0-based index of
+    /// <https://gpuweb.github.io/gpuweb/#enumdef-gpuvertexformat>.
+    /// The conversion process is described by
+    /// <https://gpuweb.github.io/gpuweb/#vertex-processing>.
+    pub format: u32,
```

Comment:
I'll do that later as part of the work to do real unpacking. It will be time consuming to do it here and without actually using it, I'll have no confidence that I have set it up correctly.